### DISCUSSION TITLE
How does typescript make generics mandatory?

### DISCUSSION THREAD

**Discussion by laterdayi** [typescript-eslint/typescript-eslint#6904] [Community Feedback]
`export const test = <T>(a:string,{c,d} = {c: T[], D:T} = {})=>{} `

test<string>()  ok
test<number>()  ok 
test()   error

**Comment by JoshuaKGoldberg** [comment]
@laterdayi it looks like you're trying to ask for help on TypeScript in general? This isn't the place for that. You might want to ask in the [TypeScript Discord](https://discord.com/invite/typescript).

Related: #6901. It's generally bad practice to file duplicates.
